src/platform/macos/system/surface.rs

```diff
@@ -123,8 +122,8 @@ impl Device {
             let size = match surface_type {
                 SurfaceType::Generic { size } => size,
                 SurfaceType::Widget { ref native_widget } => {
-                    let window: id = msg_send![native_widget.view.0, window];
-                    let bounds = window.convertRectToBacking(native_widget.view.0.bounds());
+                    let window = native_widget.view.window().unwrap();
```

Comment:
Maybe it makes sense to use `expect(...)` here as you do in places where you get the window from an NSView?